base/generator.jl

```diff
@@ -20,6 +20,12 @@ function next(g::Generator, s)
     g.f(v), s2
 end
 
+"""
+    generate(f, c...) -> iterator
+
+Apply `f` to each element of `c`.
+For multiple collection arguments, apply f elementwise.
+"""
 generate(f, c...) = Generator(f, c...)
```

Comment:
I guess, because it feels to me like it should not be necessary to think about types to use a Julia API.

It seems like there is a greater burden of understanding if I have to call a type constructor. If I'm making something explicitly this type, what else do I have to understand about this type?

If I just call a function, the type will be whatever clever type the method dispatch system figures out is best for the particular situation. Maybe `generate(f, c...)` will return a different type for some values of `f` and `c...`.

Note, that this is not actually called anywhere. It is included mostly for completeness to establish the pattern followed by `asyncgenerate` and `pgenerate`.

Also, there are precedents for having a `verb` function to create an iterator: `zip`, `filter`, `enumerate` etc.